forem/forem
# Color tokens

## What type of PR is this? (check all applicable)


## Description

In overall this is a step towards making it easier for admins to customize the look'n'feel of their Forem. This PR improves some frontend architecture parts to enable just that. 

To be more precise, this PR does few things:

1. Most importantly it introduces [color tokens and updated approach for handling colors](https://docs.google.com/document/d/14xO086NkrpqtTVOsjOKVQpF2_5o-ZEYmZu7XMWHHDSQ/edit#) (that linked doc is pretty short but it does provide a lot of context so I highly recommend reading it to better familiarize yourself with changes in this PR) in Crayons.
2. It updates some of our colors and introduces bunch of new shades that @forem/design has been requesting.
3. It significantly changes dark theme. Beyond simple color updates, it also:
    - improves a11y in few areas, i.e. links/buttons.
    - ~starts reflecting brand's accent color - so far the Dark Theme has been ignoring brand's accent color.~ **This was initially done but decided to revert it due to lots of currently unnecessary complications. [More](https://github.com/forem/forem/discussions/16161).**

### Tokens

Tokens are defined by using raw RGB values so without actually wrapping them with `rgb()` so we end up with, for example:

```
--white: 255,255,255;
```

Thanks to this we can easily reference a color in theme file but we can also easily adjust the opacity of such a color if necessary for which, so far, we've been using dedicated variables (i.e. `--accent-brand-a10` or `--base-a50`) - but that wasn't very flexible hence the change. So now we can have:

```
--btn-color: rgb(var(--white));
--btn-bg: rgba(var(--white), 0.25);
```

## Related

- [Discussion on brand colors for both themes](https://github.com/forem/forem/discussions/16161).
- [Brainstorming on overall approach](https://docs.google.com/document/d/14xO086NkrpqtTVOsjOKVQpF2_5o-ZEYmZu7XMWHHDSQ/edit).

## QA Instructions, Screenshots, Recordings

Overall this PR shouldn't really change much in light theme UI. Sure, there are bunch of updates but nothing really significant or worth "double-checking". However, it does change the dark theme quite a lot so I highly recommend playing with it and just making sure that everything still looks fine or at least not worst than how it was before :D 

It's also worth spinning up the storybook locally (`yarn storybook`) and just look through some components to verify they are still readable and visually appealing, especially on dark theme.

### UI accessibility concerns?

Nah.

## Added/updated tests?


## [Forem core team only] How will this change be communicated?

      or in a [forem.dev](http://forem.dev) post
